CMakeLists.txt

```diff
@@ -7,8 +7,11 @@ include(FindPackageHandleStandardArgs)
 # Uncomment to have the build process verbose
 # set(CMAKE_VERBOSE_MAKEFILE TRUE)
 
-# Uncomment to have the executable moved to 'build' instead of their respective 'build/xxx' directories
-#set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
+# Set output directories for libraries and executables.
+# This is important for Windows builds to have the DLLs in the same directory as the executables.
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
```

Comment:
@kozlovic Do you think this may break any existing user build scripts? As I understand, this should not affect `make install` results.

Reply:
fyi just compared make install results for this PR and main on a couple of docker container diffs and they were identical:
```
A /usr/local/include/nats.h
A /usr/local/include/nats/adapters/libevent.h
A /usr/local/include/nats/adapters/libuv.h
A /usr/local/include/nats/nats.h
A /usr/local/include/nats/status.h
A /usr/local/include/nats/version.h
A /usr/local/lib/cmake/cnats/cnats-config-release.cmake
A /usr/local/lib/cmake/cnats/cnats-config.cmake
A /usr/local/lib/libnats.so
A /usr/local/lib/libnats.so.3.9
A /usr/local/lib/libnats.so.3.9.0
A /usr/local/lib/libnats_static.a
A /usr/local/lib/pkgconfig/libnats.pc
```